Src/VimCore/RegisterMap.fs

```diff
@@ -100,9 +100,19 @@ type internal RegisterMap (_map : Map<RegisterName, Register>) =
     /// Updates the given register with the specified value.  This will also update 
     /// other registers based on the type of update that is being performed.  See 
     /// :help registers for the full details
-    member x.SetRegisterValue (reg : Register) regOperation (value : RegisterValue) = 
+    member x.SetRegisterValue (register : Option<Register>) regOperation (value : RegisterValue) (clipboardOptions : Vim.ClipboardOptions) = 

```

Comment:
I was thinking of something like `CommandUtil::SetRegisterValue (commandData : CommandData) text`.  The implementation could look at the state of `ClipboardOptions` and update both registers if necessary.  It already has access to settings so this should be doable.